src/box/applier.cc

```diff
@@ -732,8 +731,11 @@ applier_join(struct applier *applier)
 	struct iostream *io = &applier->io;
 	struct xrow_header row;
 	uint64_t row_count;
-
-	xrow_encode_join_xc(&row, &INSTANCE_UUID);
+	struct join_request req;
```

Comment:
I would rather propose to 
```
const struct join_request req = {
        .instance_uuid = INSTANCE_UUID,
        .version_id = tarantool_version_id()
}
```

Reply:
It is not possible in C++ (which is applier's language).

Reply:
oops. I tested it on MacOS and it appeared clang supports it back to `--std=c++98`, while g++ on Linux complains with `sorry, unimplemented: non-trivial designated initializers not supported`.